### ISSUE TITLE
False positive with keywords-only fields of dataclasses when using defaulted fields _and_ inheriting

### ISSUE THREAD

**Issue by huwcbjones** [pylint-dev/pylint#9040]
### Bug description

This is like #7881, but now inheritance is involved 😉 

```py
import abc, dataclasses

@dataclasses.dataclass
class Parent(abc.ABC):
    _: dataclasses.KW_ONLY
    id: int | None = None

@dataclasses.dataclass
class Child(Parent):
    name: str

c = Child(id=1, name="foo")
```

### Configuration

_No response_

### Command used

```shell
pylint --disable=all --enable=unexpected-keyword-arg mwe.py
```


### Pylint output

```shell
************* Module mwe
mwe.py:12:4: E1123: Unexpected keyword argument 'name' in constructor call (unexpected-keyword-arg)

------------------------------------------------------------------
Your code has been rated at 2.86/10 (previous run: 3.75/10, -0.89)
```


### Expected behavior

No error, because the code is fine
```py
>>> from mwe import c
>>> print(c, c.id, c.name)                                                                                                                                                                                                                                                                         
Child(id=1, name='foo') 1 foo
```

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.11.1 (main, Dec 14 2022, 20:51:11) [Clang 14.0.0 (clang-1400.0.29.202)]
```


### OS / Environment

macOS 13.5.2 (22G91)

### Additional dependencies

_No response_

**Comment by nickdrozd** [comment]
For reference: https://docs.python.org/3/library/dataclasses.html#dataclasses.KW_ONLY

**Comment by huwcbjones** [comment]
I start a new project with dataclasses...
Smack straight into "unexpected-keyword-arg"...
Search the bug tracker...
Discover I've been here before! 🤣

**Comment by jacobtylerwalls** [comment]
Removed kw-only from the reproducer:
```py
import dataclasses

@dataclasses.dataclass
class Parent():
    id: int = 1

@dataclasses.dataclass
class Child(Parent):
    name: str

c = Child(id=2, name="foo")
```
